cpp/include/games/stochastic_nim/Game.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <core/BasicTypes.hpp>
+#include <core/concepts/Game.hpp>
+#include <core/ConstantsBase.hpp>
+#include <core/GameLog.hpp>
+#include <core/GameTypes.hpp>
+#include <core/MctsConfigurationBase.hpp>
+#include <core/IOBase.hpp>
+#include <core/SimpleStateHistory.hpp>
+#include <core/TrainingTargets.hpp>
+#include <core/TrivialSymmetries.hpp>
+#include <core/WinShareResults.hpp>
+#include <games/stochastic_nim/Constants.hpp>
+#include <games/GameRulesBase.hpp>
+#include <util/EigenUtil.hpp>
+#include <util/FiniteGroups.hpp>
+#include <util/MetaProgramming.hpp>
+
+#include <boost/functional/hash.hpp>
+#include <torch/torch.h>
+
+#include <array>
+#include <cstdint>
+#include <functional>
+#include <sstream>
+#include <string>
+
+namespace stochastic_nim {
+
+struct Game {
+  struct Constants : public core::ConstantsBase {
+    using kNumActionsPerMode =
+        util::int_sequence<stochastic_nim::kMaxStonesToTake, stochastic_nim::kChanceDistributionSize>;
+    static constexpr int kNumPlayers = stochastic_nim::kNumPlayers;
+    static constexpr int kMaxBranchingFactor = stochastic_nim::kMaxStonesToTake;
```

Comment:
Technically this should be `std::max(kMaxStonesToTake, kChanceDistributionSize)`.